From df60cc3dc40b0731c298e1d61c2c6af6dfbd993a Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Thu, 30 Oct 2025 13:01:27 +0200
Subject: Unlock if crypted stop string entered

Consider the device to be in the unlocked state if the crypted stop
string has been correctly entered into the system and the autoboot has
been aborted as a result.

Upstream-Status: Pending
---
 common/autoboot.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/common/autoboot.c b/common/autoboot.c
index 5d331991c19..bcd05c84d56 100644
--- a/common/autoboot.c
+++ b/common/autoboot.c
@@ -5,6 +5,7 @@
  */
 
 #include <common.h>
+#include <unlocking.h>
 #include <autoboot.h>
 #include <bootretry.h>
 #include <cli.h>
@@ -122,6 +123,10 @@ static int passwd_abort_crypt(uint64_t etime)
 		udelay(10000);
 	} while (never_timeout || get_ticks() <= etime);
 
+	if (IS_ENABLED(CONFIG_UNLOCKING))
+		if (abort == 1)
+			device_set_unlocked();
+
 	return abort;
 }
 
-- 
2.34.1

